src/caip-types.test.ts

```diff
@@ -274,3 +278,46 @@ describe('parseCaipAccountId', () => {
     );
   });
 });
+
+describe('toCaipChainId', () => {
+  // This function relies on @metamask/utils CAIP helpers. Those are being
+  // tested with a variety of inputs.
+  // Here we mainly focus on our own wrapper around those:
+
+  it('returns a valid CAIP-2 chain ID', () => {
+    const namespace = KnownCaipNamespace.Eip155;
+    const reference = '1';
+    expect(toCaipChainId(namespace, reference)).toBe(
+      `${namespace}:${reference}`,
+    );
+  });
+
+  it.each([
+    // Too short, must have 3 chars at least
+    '',
+    'xs',
+    // Not matching
+    '!@#$%^&*()',
+    // Too long
+    'namespacetoolong',
+  ])('throws for invalid namespaces: %s', (namespace) => {
+    const reference = '1';
+    expect(() => toCaipChainId(namespace, reference)).toThrow(
+      `Invalid "namespace", must match: ${CAIP_NAMESPACE_REGEX.toString()}`,
+    );
+  });
+
+  it.each([
+    // Too short, must have 1 char at least
+    '',
+    // Not matching
+    '!@#$%^&*()',
+    // Too long
+    '012345678901234567890123456789012', // 33 chars
+  ])('throws for invalid reference: %s', (reference) => {
+    const namespace = KnownCaipNamespace.Eip155;
```

Comment:
Same point as above: does this need to be something in `KnownCaipNamespace`, or just something that matches the regex?

``` suggestion
    const namespace = 'abc';